shared/transactions/vault_create.yaml

```
@@ -0,0 +1,77 @@
+components:
+  schemas:
+    VaultCreateTransaction:
+      $id: VaultCreateTransaction
+      allOf:
+        - $ref: '../base.yaml#/components/schemas/BaseTransaction'
+      type: object
+      description: |
+        The VaultCreate transaction creates an Vault object and adds it to the relevant
+        directory node of the creator account.
+
+        If the transaction is successful, the newly created vault is owned by the account (the creator
+        account) that executed the transaction.
+      required:
+        - Asset
+      properties:
+        Asset:
+          $ref: '../base.yaml#/components/schemas/Currency'
+          description: |
+            (Required) The definition for the asset in the vault. In JSON, this is an object with currency and
+            issuer fields (omit issuer for XRP).
+        MaximumAmount:
+          type: string
+          description: |
+            (Optional) The maximum asset amount held in the vault. If not set, or set to zero, the vault will not enforce a maximum.
+        MPTokenMetadata:
+          type: string
+          format: hex
+          minLength: 1
+          maxLength: 1024
+          description: |
+            (Optional) Arbitrary metadata about this vault, in hex format. The limit for this field is 1024 bytes.
+        DomainID:
+          type: string
+          format: hash256
+          description: |
+            (Optional) The ID of the Domain ledger object for vault authorization. Can only be set if the vault is not public.
+        WithdrawalPolicy:
+          type: integer
+          format: uint8
+          description: |
+            (Optional) Numeric identifier of the vault's withdrawal policy. The only accepted input is number 1, which
+            corresponds to "first come, first served" policy.
+        Data:
+          type: string
+          format: hex
+          minLength: 1
+          maxLength: 256
```

Since a byte can accommodate 2 hex characters, the `maxLength` will be 512
```suggestion
          maxLength: 512
```